### PR TITLE
adds scriptlog endpoint

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

**Comment by drew-gross** [comment]
I think this should replace the `/logs` endpoint. We don't really want both, and since `/logs` was experimental/undocumented, we can remove it with breaking compatibility. We can add features to the `/scriptlog` endpoint compared to Parse.com, we just can't change stuff.

File: src/Routers/LogsRouter.js [hunk 1]
```diff
@@ -48,7 +43,21 @@ export class LogsRouter extends PromiseRouter {
       order,
       level,
     }
-    
+    return this.getLogs(req, options);
+  }
+  
+  handleScriptLog(req) {
+    const n = req.query.n;
```
Comment (drew-gross) [inline src/Routers/LogsRouter.js]: if you call this variable `size` you can construct the options object easier :)

Comment (flovilmart) [inline src/Routers/LogsRouter.js]: yeah...

File: src/Routers/LogsRouter.js [hunk 2]
```diff
@@ -15,11 +15,19 @@ let enforceSecurity = (auth) => {
...
       return this.handleGET(req);
     });
   }
 
+  validateRequest(req) {
+    if (!req.config || !req.config.loggerController) {
+      throw new Parse.Error(Parse.Error.PUSH_MISCONFIGURED,
+        'Logger adapter is not availabe');
+    }
+    enforceSecurity(req.auth);
```
Comment (flovilmart) [inline src/Routers/LogsRouter.js]: there is a middleware for that now on master right?

Comment (drew-gross) [inline src/Routers/LogsRouter.js]: Yep. Thanks to @nlutsenko :)

Comment (nlutsenko) [inline src/Routers/LogsRouter.js]: `middleware.promiseEnforceMasterKeyAccess` ;)

**Comment by drew-gross** [comment]
Feel free to merge with the middleware or do that later

**Pull request merged**